internal/pkg/services/argus/utils/utils.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/stackitcloud/stackit-cli/internal/pkg/errors"
+
+	"github.com/stackitcloud/stackit-sdk-go/services/argus"
+)
+
+const (
+	service = "argus"
+)
+
+func ValidatePlanId(planId string, resp *argus.PlansResponse) error {
+	if resp == nil {
+		// Should not happen, check is done before calling this function
+		return fmt.Errorf("no Argus plans provided")
+	}
+
+	for i := range *resp.Plans {
+		plan := (*resp.Plans)[i]
+		if plan.Id != nil && strings.EqualFold(*plan.Id, planId) {
+			return nil
+		}
+	}
+
+	return &errors.DSAInvalidPlanError{
+		Service: service,
+		Details: fmt.Sprintf("You provided plan ID %q, which is invalid.", planId),
+	}
+}
+
+func LoadPlanId(planName string, resp *argus.PlansResponse) (*string, error) {
+	availablePlanNames := ""
+	if resp == nil {
+		// Should not happen, check is done before calling this function
+		return nil, fmt.Errorf("no Argus plans provided")
+	}
+
+	for i := range *resp.Plans {
+		plan := (*resp.Plans)[i]
+		if plan.Name == nil {
+			continue
+		}
+		if strings.EqualFold(*plan.Name, planName) && plan.Id != nil {
```

Comment:
This is already checked in the condition above

Reply:
I think the conditions are different. Maybe there are too many checks, but the one above makes sure that plan name is not null so that the comparison is safe, then the second one checks if the id is not null so that we can return it. I used the structure from mariadb.

Reply:
Ah right, I misread it